app/src/main/java/space/taran/arknavigator/mvp/model/repo/PreviewRepo.kt

```diff
@@ -0,0 +1,56 @@
+package space.taran.arknavigator.mvp.model.repo
+
+import space.taran.arknavigator.mvp.model.dao.ResourceId
+import space.taran.arknavigator.mvp.model.dao.computeId
+import space.taran.arknavigator.ui.fragments.utils.PredefinedIcon
+import space.taran.arknavigator.ui.fragments.utils.Preview
+import space.taran.arknavigator.utils.*
+import java.nio.file.Files
+import java.nio.file.Path
+
+class PreviewRepo {
+
+    fun providePreview(path: Path, resourceID: ResourceId? = null): Preview {
+        if (Files.isDirectory(path)) {
+            return Preview(predefinedIcon = PredefinedIcon.FOLDER)
+        }
+
+        return createPair(path, resourceID)
+    }
+
+    private fun createPair(file: Path, resourceID: ResourceId?): Preview {
+        return when {
+            isImage(file) -> {
+                Preview(
+                    fileType = FileType.IMAGE,
+                    previewPath = file,
+                    fileExtension = extensionWithoutDot(file)
+                )
+            }
+            isVideo(file) -> {
+                Preview(
+                    fileType = FileType.VIDEO,
+                    previewPath =file,
+                    fileExtension = extensionWithoutDot(file),
+                    //extraInfo = getVideoInfo(file)
+                )
+            }
+            isPDF(file) -> getPdfPreview(file, resourceID)
+            isFormat(file, ".gif") ->
+                Preview(
+                    fileType = FileType.GIF,
+                    previewPath = file,
+                    fileExtension = extensionWithoutDot(file)
+                )
+            else -> Preview(fileExtension = extensionWithoutDot(file))
+        }
+    }
+
+    private fun getPdfPreview(file: Path, resourceID: ResourceId? = null): Preview {
```

Comment:
Not sure how you want to use Path for this part. Here, we compare the id of the resource to the list of saved previews. When a preview is saved, it's saved using it's calculated id. In this part, we see if there is a file with a name that matches the ID. I understand that we need to remove our dependency on the ID, since it's unreliable. what should be the naming convention for previews, so we can save and retrieve them efficiently?